### ISSUE TITLE
New Lua errors since last update on all toons and Yes they are level 110

### ISSUE THREAD

**Issue by Jefferyasbell** [adde88/Nerdpack-Zylla#26]
4x Nerdpack-Zylla\classes.lua:307: attempt to index local 'spec' (a nil value)
Nerdpack-Zylla\classes.lua:307: in function <Nerdpack-Zylla\classes.lua:305>
(tail call): ?
Nerdpack-Zylla\fakeunits.lua:6: in function <Nerdpack-Zylla\fakeunits.lua:5>
Nerdpack-Zylla\fakeunits.lua:92: in function `?'
NerdPack\system\fakeunits.lua:28: in function <NerdPack\system\fakeunits.lua:24>
NerdPack\system\fakeunits.lua:53: in function <NerdPack\system\fakeunits.lua:41>
NerdPack\system\fakeunits.lua:65: in function `Process'
NerdPack\system\fakeunits.lua:50: in function <NerdPack\system\fakeunits.lua:41>
NerdPack\system\fakeunits.lua:65: in function `Process'
NerdPack\system\fakeunits.lua:74: in function `Filter'
NerdPack\system\parser.lua:95: in function <NerdPack\system\parser.lua:93>
(tail call): ?
NerdPack\system\parser.lua:111: in function `func'
NerdPack\system\parser.lua:101: in function <NerdPack\system\parser.lua:93>
(tail call): ?
NerdPack\system\parser.lua:195: in function <NerdPack\system\parser.lua:180>
SharedXML\C_TimerAugment.lua:16: in function <SharedXML\C_TimerAugment.lua:14>

Locals:
InCombatSkipped

I get errors on all toons that I use Zylla for CR Like this one. I just downloaded the update today a few min ago.

**Comment by adde88** [comment]
Debugging as we speak :)

**Comment by adde88** [comment]
This is happening because you're using one of the "FakeUnits" in Settings, and a function within WoW that should report your SpecID is not returning your SpecID, but it's returning 0.
I've made a ticker that checks this value every second.
Which should fix the problem.
If it happens again, just /reload
I'll see if i can add another better fix in the future.

**Comment by Jefferyasbell** [comment]
Ok ...will do

**Comment by Jefferyasbell** [comment]
Ok I just was playing my DH. I have not played it since I posted the lua error a few days ago and I am getting the same lua error that I got before....I dont see in the options in the Zilla's combat Routines an option to toggle FakeUnits on of off, where do I turn it off. I also looked in the nerdpack settings and dont see an option there either. I love the CR you have created. I Just wish I could use it.

this is the lua error I am getting

7x Nerdpack-Zylla\classes.lua:307: attempt to index local 'spec' (a nil value)
Nerdpack-Zylla\classes.lua:307: in function <Nerdpack-Zylla\classes.lua:305>
(tail call): ?
Nerdpack-Zylla\fakeunits.lua:6: in function <Nerdpack-Zylla\fakeunits.lua:5>
Nerdpack-Zylla\fakeunits.lua:92: in function `?'
NerdPack\system\fakeunits.lua:28: in function <NerdPack\system\fakeunits.lua:24>
NerdPack\system\fakeunits.lua:53: in function <NerdPack\system\fakeunits.lua:41>
NerdPack\system\fakeunits.lua:65: in function `Process'
NerdPack\system\fakeunits.lua:50: in function <NerdPack\system\fakeunits.lua:41>
NerdPack\system\fakeunits.lua:65: in function `Process'
NerdPack\system\fakeunits.lua:74: in function `Filter'
NerdPack\system\parser.lua:95: in function <NerdPack\system\parser.lua:93>
(tail call): ?
NerdPack\system\parser.lua:111: in function `func'
NerdPack\system\parser.lua:101: in function <NerdPack\system\parser.lua:93>
(tail call): ?
NerdPack\system\parser.lua:195: in function <NerdPack\system\parser.lua:180>
SharedXML\C_TimerAugment.lua:16: in function <SharedXML\C_TimerAugment.lua:14>

Locals:
InCombatSkipped

**Comment by Jefferyasbell** [comment]
I think I may have found error If I select nearest in the targeting I get the error if I select normal I dont get an error. so That must be the toggle you mentioned....One quick question...When I go to a target dummy and try to test my dps It wont attack on either of my toons that are 110. I have selected the Pref tallents and nothing. If I go out and attack a mob in game it is fine it auto attacks. Is this working as intended or is it a bug.

**Comment by adde88** [comment]
Those different targets you can select are the FakeUnits.
Using "Normal" will simply attack your current target.
I've been testing this quite thoroughly on both my Mage and Demon Hunter now.
I've also sent another update, that hopefully will fix the problem you had with the LUA error.

When you're trying to attack a dummy; are you using "Normal" as the selected target in the Settings?

**Comment by Jefferyasbell** [comment]
Yes I was. I checked and was on normal....Do I need to change that?

**Comment by adde88** [comment]
Nah Normal is the same as using "target". Which should work on dummies as it doesn't use any custom unit stuff.
